### ISSUE TITLE
error xxxDbEntity is not managed while calling loadItems() on a collection field of a managed entity 

### ISSUE THREAD

**Issue by BrannJoly** [mikro-orm/mikro-orm#5082]
### Describe the bug

Hi,
Thank you very much for the great work on v6, lots of good stuff I was really looking forward to!

After upgrading to v6, my tests fail with errors
` rollback Entity xxx is not managed. An entity is managed if its fetched from the database or registered as new through EntityManager.persist()`

My code used to work fine with v5 (no error, and no runtime bugs)

I didn't see anything about this change in the blog post nor in the migration date.

### Reproduction

see reproductiuon repo [here](https://github.com/BrannJoly/mikro-orm-bug-1) : 

relevant excerpt from the repo

``` javascript

@Entity()
export class User {
....
 @ManyToMany(() => Item)
 items = new Collection<Item>(this);
}

test('bug', async () => {

  const u = new User("foo","foo@x.com")
  orm.em.persist(u)
  await u.items.loadItems() // error : Entity User is not managed. An entity is managed if its fetched from the database or registered as new through EntityManager.persist()
  
});

```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.0

### Node.js version

18.16.0

### Operating system

osX

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by BrannJoly** [comment]
Well, at the very least the error message is misleading (it asks me to do an  EntityManager.persist() which I just did on the previous line)

Obviously my code doesn't make sense, but I needed a minimal reproduction. 

My actual situation: 
I've some mapping code between dtos and dbEntities. I've mutualized the code used to create a dbEntity and to update it. The mapping code is not completely trivial as I have deeply nested objects with potential cyclical relationships

So far, I was trying to loadItems(), and it would do nothing on a new entity, and load the actual entities if the entity was preexisting.
With the latest version, instead of doing nothing, loadItem() now fails, whcih leads to a regression in perefectly working code. I think it might make sense to add a note in the docs maybe ?

I was looking for a way to test if the entity was managed before calling loadItems(), something like if (entity.isManaged()) but I didn't find anything. Is there an easy way to do that?

Alternatively, I can probably refactor my code to deal with the issue, but I thought it was worth reporting, at least so that the error message can be improved!

**Comment by B4nan** [comment]
The error message is kinda correct, your entity is not managed, it becomes managed after flushing, not on persist call, but I get what you mean, we can improve the suggestion to mention the flushing too.

I'll have a look at how this behaves in v5, as I am really curious :]

> So far, I was trying to loadItems(), and it would do nothing on a new entity, and load the actual entities if the entity was preexisting.

Hmm it's true that a new entity should have initialized collections so the loadItems call should just resolve to empty array right ahead. Maybe that's how/why it works in v5.

**Comment by BrannJoly** [comment]
I've temporarily fixed my codebase with v6 by wrapping the loadItems() in a try catch.
If there is a way to know if my collection is managed, I would prefer using a test rather than resorting to an exception ...
thanks !

**Comment by B4nan** [comment]
An entity is managed when it has a reference to the EM that is managing it, so one way is to check `wrap(e, true).__em`, it will be undefined for not managed ones. Then there is also `__managed` boolean, but not 100% sure now if it can be used this way too (probably yes). We can surely add a public getter that will just check the `__em` prop.

**Comment by B4nan** [comment]
I messed up the commit message, but this is fixed via 3b94bf957243e8ded787c853915c84cfc94832d7

**Comment by BrannJoly** [comment]
amazing ! thanks a lot for the extra quick turnaround ! :)
